test/evergreen/print_stack_trace.py

```diff
@@ -51,20 +51,25 @@ def _find_debugger(debugger: str):
         """Find the installed debugger."""
         return which(debugger)
 
-    def dump(self, exe_path: str, core_path: str):
+    def dump(self, exe_path: str, core_path: str, dump_all: bool, output_file: str):
         """Dump stack trace."""
         if self.dbg is None:
             sys.exit("Debugger lldb not found,"
                      "skipping dumping of {}".format(core_path))
 
-        cmds = [
-            "thread backtrace all -c 30",
-            "quit"
-        ]
-
+        cmds = []
+        if dump_all:
+            cmds.append("thread apply all backtrace -c 30")
+        else:
+            cmds.append("backtrace -c 30")
+        cmds.append("quit")
+
+        output = None
+        if (output_file):
+            output = open(output_file, "w")
         subprocess.run([self.dbg, "--batch"] + [exe_path, "-c", core_path] +
                        list(itertools.chain.from_iterable([['-o', b] for b in cmds])),
-                       check=True)
+                       check=True, stdout=output)
```

Comment:
If output equals None (e.g. no output_file provided), stdout won't be captured. Just to confirm that's the expected behavior.

Reply:
Unfortunately, I don't have background information on this.
The following code works as expected though:
```
            dbg = GDBDumper()
            dbg.dump(executable_path, core_file_path, lib_path, False, None)

            # Extract the filename from the core file path, to create a stacktrace output file.
            file_name, _ = os.path.splitext(os.path.basename(core_file_path))
            dbg.dump(executable_path, core_file_path, lib_path, True, file_name + ".stacktrace.txt")
```
We first print the stack trace in the Evergreen logs and then in a file.

Reply:
It looks like WT-8328 initially introduced this change. @jiechenbo do you mind sharing the context?

Reply:
hey sorry, I must've missed this. I forgot the context for this python file unfortunately. Briefly looking at the code it does seem that we still follow the expected behaviour @lukech. If an output_file is not specified stdout is not captured, if that is what we are confirming?

Reply:
Sorry I misinterpreted the question, yes @lukech that is the expected behaviour.

Reply:
Sorry I have misinterpret the question, Yes @etienneptl is correct, the context is that when stdout=None, it means that the subprocess will run with default stdout and stderr pipelines. Meaning that the subprocess would output the logs onto the evergreen logs. When stdout is pointed to a file, it redirects all its output into the file instead.

Reply:
According to [subprocess documentation](https://docs.python.org/3/library/subprocess.html), "None if stdout was not captured." Does it indicate `stdout=None` means the stdout won't be captured at all (neither in Evergreen logs nor in a file)?

Reply:
Hey @lukech the documentation is very confusing here. So that documentation refers to when a `subprocess.CompletedProcess` and refers to stdout not being captured when parsing through the process output. But if you pass it inside inside run(), that means stdout will act as normal if set to None
```
With the default settings of None, no redirection will occur; the child’s file handles will be inherited from the parent.
```